src/DisputeHandler.ts

```diff
@@ -55,164 +72,281 @@ class DisputeHandler {
+        proofs: ProofStruct[],
+        timestamp: BigNumberish,
+        timeout?: TimeoutStruct
+    ): Promise<{
+        dispute: DisputeStruct;
+        disputeAuditingData: DisputeAuditingDataStruct;
+    }> {
+        const disputeIndex =
+            await this.stateChannelManagerContract.getDisputeLength(
+                this.channelId
+            );
+        const genesisStateSnapshotHash =
+            this.agreementManager.getForkGenesisSnapshot(
+                forkCnt
+            )!.stateMachineStateHash;
+        const latestStateSnapshotHash = this.agreementManager.getSnapShot(
+            forkCnt,
+            transactionCnt
+        )!.stateMachineStateHash;
+        const stateProof = this.collectStateProof(forkCnt);
+        const onChainSlashes =
+            await this.stateChannelManagerContract.getOnChainSlashedParticipants(
+                this.channelId
+            );
+        const onChainLatestJoinChannelBlockHash =
+            this.agreementManager.getLatestJoinChannelBlockHash();
+        const latestExitChannelBlockHash =
+            this.agreementManager.getLatestJoinChannelBlockHash();
+        const exitChannelBlocks = this.agreementManager.getExitChannelChain();
 
-        const forkCntNumber = Number(forkCnt);
-        //TODO! stop signing for the current fork
-        this.setForkDisputed(forkCntNumber);
-        proofs.forEach((proof) => this.addProof(forkCntNumber, proof));
-        const _dispute = this.disputes.get(forkCntNumber);
-        if (!_dispute) {
-            await this.createNewDispute(
+        const {
+            encodedModifiedState,
+            exitBlock,
+            totalDeposits,
+            totalWithdrawals
+        } = (await this.stateChannelManagerContract.generateDisputeOutputState(
+            this.agreementManager.getStateMachineState(
                 forkCnt,
-                foldedParticipant,
-                foldedTransactionCnt,
-                proofs
-            );
-        }
+                transactionCnt
+            )!,
+            proofs,
+            { channelId: this.channelId },
+            onChainSlashes,
+            ethers.ZeroAddress,
+            ethers.ZeroAddress,
+            this.agreementManager.getJoinChannelChain(),
+            this.agreementManager.getSnapShot(forkCnt, transactionCnt)!
+        )) as unknown as DisputeOutputState;
+
+        const participants =
+            (await this.stateChannelManagerContract.getStatemachineParticipants(
+                encodedModifiedState
+            )) as unknown as AddressLike[];
 
-        const newDispute = await this.stateChannelManagerContract.getDispute(
-            this.channelId
+        const disputeOutputStateSnapshot: StateSnapshotStruct = {
+            stateMachineStateHash: ethers.keccak256(
+                ethers.toUtf8Bytes(encodedModifiedState)
+            ),
+            participants,
+            forkCnt: forkCnt + 1,
+            latestExitChannelBlockHash,
```

Comment:
This is the result from applying slashes and removals during generateDisputeOutputState, but you don't use that - it's not possible to know this from agreementManager when creating the dispute